dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/CrudControllerAdvice.java

```diff
@@ -489,14 +489,34 @@ public void servletExceptionHandler(ServletException ex) throws ServletException
     throw ex;
   }
 
-  @ExceptionHandler({
-    IllegalArgumentException.class,
-    SchemaPathException.class,
-    JsonPatchException.class
-  })
+  @ExceptionHandler({SchemaPathException.class, JsonPatchException.class})
+  @ResponseBody
+  public WebMessage handleBadRequest(Exception ex) {
+    return badRequest(ex.getMessage());
+  }
+
+  /**
+   * Handles {@link IllegalArgumentException} and prints the stack trace to standard error. {@link
+   * IllegalArgumentException} is used in DHIS 2 application code but also by various frameworks to
+   * indicate programming errors, so stack trace must be printed and not swallowed.
+   */
+  @ExceptionHandler(IllegalArgumentException.class)
   @ResponseBody
-  public WebMessage handleBadRequest(Exception exception) {
-    return badRequest(exception.getMessage());
+  public WebMessage illegalArgumentExceptionHandler(IllegalArgumentException ex) {
+    ex.printStackTrace();
```

Comment:
Is it better to use the logging framework for this? If the log output is not the same, some log consumers might have trouble parsing it.

Reply:
Yes agreed. I did it like this for consistency, as we use `ex.printStackTrace();` for the catch-all error handler `defaultExceptionHandler` further down. Happy to change it. Though we should probably do the same in all these three methods. Thoughts @netroms @jbee ?

Reply:
Yeah, I think we should never use e.printStackTrace(), since it will always print directly to stdout, unformatted.

Reply:
That's okay. Updated it now. All three places now use `log.error`.